internal/app/maintenance/server/server.go

```diff
@@ -201,3 +204,64 @@ func (s *Server) Upgrade(ctx context.Context, in *machine.UpgradeRequest) (reply
 
 	return reply, nil
 }
+
+// Reset resets the node.
+//
+//nolint:gocyclo
+func (s *Server) Reset(ctx context.Context, in *machine.ResetRequest) (reply *machine.ResetResponse, err error) {
+	if err = assertPeerSideroLink(ctx); err != nil {
+		return nil, err
+	}
+
+	actorID := uuid.New().String()
+
+	log.Printf("reset request received. actorID: %s", actorID)
+
+	if len(in.GetSystemPartitionsToWipe()) > 0 {
+		return nil, fmt.Errorf("system partitions to wipe params is not supported in the maintenance mode")
+	}
+
+	var dev *blockdevice.BlockDevice
+
+	disk := s.controller.Runtime().State().Machine().Disk(disk.WithPartitionLabel(constants.BootPartitionLabel))
+
+	if disk == nil {
+		return nil, fmt.Errorf("reset failed: Talos is not installed")
+	}
+
+	dev, err = blockdevice.Open(disk.Device().Name())
```

Comment:
should we also support user disks here? this might be needed to get the machine into a "clean" state

Reply:
good idea, added!